internal/jenkinsapi/jenkinsapi.go

```diff
@@ -31,32 +32,55 @@ func NewJenkinsAPI(tenant clients.TenantService, idler clients.IdlerService) Jen
 	}
 }
 
-// Start returns the Jenkins status for the current user
-func (api *jenkinsAPIImpl) Start(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+// Returns namespace from request headers
+func fetchNamespace(w http.ResponseWriter, r *http.Request, t clients.TenantService) (clients.Namespace, error) {
 	resp := clients.StatusResponse{}
 
 	authHeader := r.Header.Get("Authorization")
 	if !strings.HasPrefix(authHeader, "Bearer ") {
-		HandleError(w, resp, errors.New("Could not find Bearer Token in Authorization Header"), http.StatusUnauthorized)
-		return
+		HandleError(w, resp, errors.New("Could not find Bearer token in Authorization Header"), http.StatusUnauthorized)
 	}
-
-	accessToken := strings.Split(r.Header.Get("Authorization"), " ")[1]
-	namespace, err := api.tenant.GetNamespace(accessToken)
+	accessToken := strings.Split(authHeader, " ")[1]
+	namespace, err := t.GetNamespace(accessToken)
 	if err != nil {
 		HandleError(w, resp, err, http.StatusUnauthorized)
-		return
+		return namespace, err
 	}
 	log.Infof("Found token info in the query. Namespace is %s and clusterURL is %s", namespace.Name, namespace.ClusterURL)
+	return namespace, nil
+}
+
+// Queries Idler state and returns it
+func fetchState(w http.ResponseWriter, r *http.Request, namespace clients.Namespace, idler clients.IdlerService) (clients.PodState, error) {
+	resp := clients.StatusResponse{}
 
-	status, err := api.idler.State(namespace.Name, namespace.ClusterURL)
+	status, err := idler.State(namespace.Name, namespace.ClusterURL)
 	if err != nil {
 		HandleError(w, resp, err, http.StatusInternalServerError)
-		return
 	}
+
+	return status, err
+}
+
+// Returns the Jenkins pods status for current user
+func (api *jenkinsAPIImpl) Status(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	resp := clients.StatusResponse{}
+
+	namespace, _ := fetchNamespace(w, r, api.tenant)
+	status, _ := fetchState(w, r, namespace, api.idler)
```

Comment:
I am not sure if need to refactor out fetchState since it doesn't seem to save us anything if we move the error handling to the caller.